src/Microsoft.Health.Fhir.Shared.Api/Features/Resources/Bundle/BundleHandler.cs

```diff
@@ -179,7 +184,16 @@ private async Task ExecuteRequests(Hl7.Fhir.Model.Bundle responseBundle, Hl7.Fhi
                     entryComponent.Response = new Hl7.Fhir.Model.Bundle.ResponseComponent { Status = ((int)HttpStatusCode.NotFound).ToString() };
                 }
 
-                responseBundle.Entry.Add(entryComponent);
+                if (responseBundle.Entry?.Any() != true)
+                {
+                    List<EntryComponent> entryComponents = new EntryComponent[entrySize].ToList();
+                    entryComponents[entry.Value] = entryComponent;
+                    responseBundle.Entry = entryComponents;
```

Comment:
This could be simplified to `responseBundle.Entry = new List<EntryComponent>(entrySize)`

Reply:
I wrote this first too and found out that it creates a list and assigns the capacity to entrySize but the list itself is empty ie. entry can't be added to list[index]. So with the above logic, if Entry is null or count is zero, I am creating a list with the rows to the length of entrySize and setting it to responseBundle. Then for the subsequent iteration only else part will be executed

Reply:
Adding a comment in the code could help whoever read the code later to understand why it's done this way.